### PR TITLE
Performance fix for AnalysisEntityMapAbstractDomain.Merge algorithm

### PR DESCRIPTION
**Pull request by mavasani**
We now ensure that we only create new entries in the result map if the entity key is a valid tracked entity

### CODE REVIEW

File: src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PointsToAnalysis/TrackedEntitiesBuilder.cs [hunk 1]
```diff
@@ -10,16 +12,29 @@ namespace Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.PointsToAnalysis
...
         public void Dispose()
         {
-            AllEntities.Free();
+            _allEntities.Free();
         }
+
+        public void AddEntity(AnalysisEntity analysisEntity)
+            => _allEntities.Add(analysisEntity);
+
+        public IEnumerable<AnalysisEntity> EnumerateEntities()
```
Comment (dotpaul) [inline src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PointsToAnalysis/TrackedEntitiesBuilder.cs]: >EnumerateEntities [](start = 43, length = 17)

Nit: Could make TrackedEntitiesBuilder implement IEnumerable<AnalysisEntity>.

Comment (mavasani) [reply src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PointsToAnalysis/TrackedEntitiesBuilder.cs]: This is no longer applicable as now this type tracks both entities and points to values.

**Comment by mavasani** [comment]
@dotpaul I was able to validate no performance regression in building Roslyn after this fix, let me know once your validations are done. Thanks!

**Comment by dotpaul** [comment]
Other than the one case @LLLXXXCCC emailed about, the previous repros looks good to me! I'm trying a pass on a larger code base; will approve after that finishes.

**Comment by mavasani** [comment]
Thank you @dotpaul!

**Pull request merged**